crates/bevy_pbr/src/light/mod.rs

```diff
@@ -708,77 +711,88 @@ pub fn check_dir_light_mesh_visibility(
                 .entry(*view)
                 .or_insert_with(|| vec![VisibleEntities::default(); frusta.len()]);
         }
+
         for v in views_to_remove {
             visible_entities.entities.remove(&v);
         }
 
         // NOTE: If shadow mapping is disabled for the light then it must have no visible entities
         if !directional_light.shadows_enabled || !light_view_visibility.get() {
-            continue;
+            return;
         }
 
         let view_mask = maybe_view_mask.unwrap_or_default();
 
-        for (
-            entity,
-            inherited_visibility,
-            mut view_visibility,
-            maybe_entity_mask,
-            maybe_aabb,
-            maybe_transform,
-            has_visibility_range,
-        ) in &mut visible_entity_query
-        {
-            if !inherited_visibility.get() {
-                continue;
-            }
-
-            let entity_mask = maybe_entity_mask.unwrap_or_default();
-            if !view_mask.intersects(entity_mask) {
-                continue;
-            }
+        for (view, view_frusta) in &frusta.frusta {
+            visible_entity_query.par_iter().for_each_init(
+                || {
+                    let mut entities = view_visible_entities_queue.borrow_local_mut();
+                    entities.resize(view_frusta.len(), Vec::default());
+                    (visible_entities_queue.borrow_local_mut(), entities)
+                },
+                |(queue0, queue1),
```

Comment:
I think either some descriptive names for these queues, or a short comment about what they're used for, would help readability. When skimming I first thought both contained entities because of the similar names.